transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -389,13 +393,47 @@ protected boolean offerChannel(Channel channel) {
 
     @Override
     public void close() {
+        closeIdleChannels(nextEventLoop());
+    }
+
+    void closeIdleChannels(EventExecutor executor) {
+        Promise<Void> result = executor.newPromise();
+        closeIdleChannels(result);
+        if (!executor.inEventLoop()) {
+            result.syncUninterruptibly();
```

Comment:
Sync uninterruptibly is not a good idea.  When the program is trying to shutdown (which is presumably when this method would be called), getting an Interruption means give up.  This should either declare it throws IE or wrap it in a runtime exception.

Reply:
This method could exit before all the channels are closed if it is responsive to interrupts. Event loop threads will still be running even though the current thread is free to go. I'm not sure if it's okay or not. Re-interrupt + runtime exception would be the best option wrt to backward compatibility - in my opinion.

Reply:
SGTM